src/util/test/MetricTests.cpp

```diff
@@ -28,6 +28,31 @@ using uniform_u64 = stellar::uniform_int_distribution<uint64_t>;
 // how much data to keep in memory when comparing datasets
 static std::chrono::seconds const sampleCutoff(60 * 5);
 
+void
+sleepTillNextBucketIfNecessary()
+{
+    // Medida doesn't support any virtual clock.
+    // Therefore, the following tests use the real time.
+    // This makes tests flaky as they will fail when it's too close to the end
+    // of a bucket. For instance, if a test starts at 11:59:59.998, then the
+    // first event it records might go into [11:59:30, 12:00:00] but the next
+    // event might go into [12:00:00, 12:00:30] even if they occur within 0.003
+    // seconds. This function checks if we're near the end of a time bucket, and
+    // if so it sleeps for a little bit to ensure that we're past that.
+
+    auto now = std::chrono::system_clock::now().time_since_epoch();
+    long long nowInMs =
+        std::chrono::duration_cast<std::chrono::milliseconds>(now).count();
+    if (nowInMs % 1000 > 900)
+    {
+        // We are less than 100 seconds away from the next second.
+        // We might be too close to the end of the time bucket if this is the
+        // last second. Sleep for 150 milliseconds to ensure that we will
```

Comment:
it should probably be
* a while loop that sleeps for a shorter time instead of sleeping for 150ms. The way "sleep" works there is really no guarantee on how long it sleeps, just that it's "at least" what is passed in, this is especially true when there are multiple threads (like in Github where we run tests in parallel)
* I would wait to be in the first 10% (=100 ms )of the window just to be sure (instead of the first 900 ms like now)

Reply:
Fixed. It sleeps for 50 ms until it's in the first 100 ms. I also updated the window size to 5 sec to further reduce the chance of accidentally going to the next time window.